pkg/util/push/push.go

```diff
@@ -67,43 +67,15 @@ func handler(maxRecvMsgSize int,
 				logger = log.WithSourceIPs(source, logger)
 			}
 		}
-		bufHolder := bufferPool.Get().(*bufHolder)
-		var req mimirpb.PreallocWriteRequest
-		buf, err := parser(ctx, r, maxRecvMsgSize, bufHolder.buf, &req)
-		if err != nil {
-			level.Error(logger).Log("err", err.Error())
 
-			// Check for httpgrpc error.
-			if resp, ok := httpgrpc.HTTPResponseFromError(err); ok {
-				http.Error(w, string(resp.Body), int(resp.Code))
-			} else {
-				http.Error(w, err.Error(), http.StatusBadRequest)
-			}
-
-			bufferPool.Put(bufHolder)
-			return
-		}
-		// If decoding allocated a bigger buffer, put that one back in the pool.
-		if len(buf) > len(bufHolder.buf) {
-			bufHolder.buf = buf
-		}
-
-		cleanup := func() {
-			mimirpb.ReuseSlice(req.Timeseries)
-			bufferPool.Put(bufHolder)
+		req := &Request{
```

Comment:
would it make sense to pool these `Request`s?

Reply:
I don't think we should do that without any profiling first.

Reply:
@replay are you ok if I skip this from this PR?

Reply:
Sure I'm ok with that, I meant it like a genuine question :)